wasm/src/main/scala/ir2wasm/WasmExpressionBuilder.scala

```diff
@@ -1735,4 +1738,30 @@ private class WasmExpressionBuilder private (
 
     IRTypes.AnyType
   }
+
+  private def genClone(t: IRTrees.Clone): IRTypes.Type = {
+    val expr = fctx.addSyntheticLocal(TypeTransformer.transformType(t.expr.tpe)(ctx))
+    fctx.block(Types.WasmRefType(Types.WasmHeapType.ObjectType)) { ourObject =>
+      genTreeAuto(t.expr)
+      instrs += BR_ON_CAST(
+        CastFlags(false, false),
+        ourObject,
+        HeapType(Types.WasmHeapType.Simple.Any),
+        HeapType(Types.WasmHeapType.ObjectType)
+      )
+      instrs += UNREACHABLE
+    }
+    instrs += LOCAL_TEE(expr)
+    instrs += REF_AS_NOT_NULL // cloneFunction argument is not nullable
+
+    instrs += LOCAL_GET(expr)
+    instrs += STRUCT_GET(TypeIdx(WasmStructTypeName(IRNames.ObjectClass)), StructFieldIdx.vtable)
+    instrs += STRUCT_GET(
+      TypeIdx(WasmTypeName.WasmStructTypeName.typeData),
+      WasmFieldName.typeData.cloneFunctionIdx
+    )
+    // cloneFunction: (ref j.l.Object) -> ref j.l.Object
+    instrs += CALL_REF(TypeIdx(WasmTypeName.WasmFunctionTypeName.cloneFunction))
+    t.tpe
```

Comment:
You may have to `REF_CAST` the result of `cloneFunction` back to `t.tpe`. If called on a specific subclass of `Cloneable`, `t.tpe` will be that subclass, and hence the subsequent code may expect something of the right subtype.

Reply:
https://github.com/tanishiking/scala-wasm/pull/36/commits/15941fc6d2ac33977473948fc1f4ba70bdb3870c